users/user.rb

```diff
@@ -2,7 +2,10 @@
 class User
     # include page_collection do i need this? require/include..
     # i think i dont but lets see
-    attr_reader :current_page
+
+    # andrew: I included the page_builder as a read only class variable. Is there a better way
+    #         of going about this? Got confused when adding the 'add home' functionality
+    attr_reader :current_page, :page_builder
```

Comment:
Similar to above, where did you need to read these from that they needed to be added as attr_reader's ?

Reply:
The more I think about it, for like single action tasks the workflow will have very little in it.
When they get more complicated, there will be more here.
IE for something like creating a new home and setting up an appointment,
it might just be 
user.create_home
user.create_new_source
user.schedule_appointment

and thats really it.
Almost to the point where if we wanted to just chain smaller ones together, we should just make them smaller
Maybe we've got one too many layers of abstraction lol
(as in maybe the logic should just go here in the workflow, give it access the user and everything it needs)
I'll think about it tomorrow while I'm not so sleepy